### ISSUE TITLE
Autocomplete of suggested argument is not working with return key

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#44570]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1044138/autocomplete-of-suggested-argument-is-not-working.html)._

---
Autocomplete of suggested argument is not working with return key.

I have this class
```
public class ProductsController : ControllerBase
    {
        public ProductsController()
        {
        }
    }
```

Now, in the controller I add the file type `JsonFileProductService` with the argument name `pro` and I get suggestions for the autocomplete.
The highlighted suggested name is `productService`. When I click the return key on my keyboard, it doesn't autocomplete it, instead it writes a new line. I have to use the tab key here to actually autocomplete it.

For example when I write `Console.Wr` I get the autocomplete suggestion `WriteLine`. When I press the return key, it writes `Console.WriteLine`.

---
#### Visual Studio Feedback System on 5/22/2020, 03:30 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by Zerotask** [comment]
I made a little typo
> Now, in the controller

I meant the constructor.

**Comment by CyrusNajmabadi** [comment]
This is by design.  these items are 'soft selected' so that they don't interfere with the names a user may be trying to write themselves.  i.e. imagine that someone wanted to write:

```c#
JsonFileProductService pro
    = ...
```

So they type `pro<enter>`.  If we commit our item we will override their explicitly chosen name.  That's why we make it so that you have to explicitly choose the item (with `<tab>` or arrow keys).

**Comment by Zerotask** [comment]
@CyrusNajmabadi Thanks for the input.
In this case I'd click ESC key to cancel the suggestion and then continue, but I get your point.

So the decision here is how likely your scenario is compared to the used clicking the return key for autocompletion.

**Comment by CyrusNajmabadi** [comment]
> So the decision here is how likely your scenario is compared to the used clicking the return key for autocompletion.

Yes.  We have experimented with more 'aggressive' forms in the past, where 'enter' could capture stuff like this, and it was met with an extremely negative reaction.   We think it's far more preferable to have some users be more explicit that they want the suggestion then it is to have others have to opt-out of the suggestion. 

Tab has been the universal 'commit' key for completion since the first version of VS.  So we've kept that as the easy thing to learn and use to commit things, and we use suggestion-mode to provide these options without interfering with people who don't always want to use them.
